Framework/NexusGeometry/src/NexusGeometryParser.cpp

```diff
@@ -595,6 +602,109 @@ void parseMonitors(const H5::Group &root, InstrumentBuilder &builder) {
   }
 }
 
+class Tube {
+private:
+  Eigen::Vector3d axis;
+  double height = 0;
+  double radius;
+  std::vector<Eigen::Vector3d> m_positions;
+  std::vector<int> m_detIDs;
+  Eigen::Vector3d p1;
+  Eigen::Vector3d p2;
+
+public:
+  const Eigen::Vector3d &location() const { return m_positions[0]; }
+  const std::vector<Eigen::Vector3d> &positions() const { return m_positions; }
+  const std::vector<int> &detIDs() const { return m_detIDs; }
+
+  boost::shared_ptr<const Mantid::Geometry::IObject> shape() {
+    Eigen::Matrix<double, 3, 3> points;
+    // Centre shape about (0, 0, 0)
+    auto p1 = m_positions.front();
+    auto p2 = m_positions.back();
+    auto centre = (p1 + p2) / 2;
+
+    points.col(0) = p1 - centre;
+    points.col(2) = p2 - centre;
+
+    // Find point on outer circle
+    auto norm = axis.norm();
+    auto factor = radius / norm;
+    auto pointOnCircle = (axis * factor) + (p1 - centre);
+    points.col(1) = pointOnCircle;
+
+    return NexusShapeFactory::createCylinder(points);
+  }
+
+  Tube(Eigen::Vector3d axis, double radius, Eigen::Vector3d pos, int detID) {
```

Comment:
Inputs here need explaining. For example, the magnitude of the axis is important in the calculations (gives pixel height), so must be un-normalised. `pos` needs description.

Reply:
Wold be nicer/faster if we knew the approximate size of the tube up front. Is that possible?